lib/spack/spack/spec.py

```diff
@@ -3181,7 +3171,7 @@ def _evaluate_dependency_conditions(self, name):
         for when_spec, dependency in conditions.items():
             if self.satisfies(when_spec):
                 if dep is None:
-                    dep = dp.Dependency(self.name, Spec(name), depflag=0)
+                    dep = dp.Dependency(Spec(self.name), Spec(name), depflag=0)
```

Comment:
Even here it seems safe to use a copy instead of `self`. Reviewing this code makes me appreciate the new concretizer more.

Reply:
This change was really just to ensure early `Spec` parsing (and thus validation). It didn't have a self ref before -- just the name.